runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessInstanceEvent/ProcessInstanceEvent.reducers.ts

```diff
@@ -18,3 +20,27 @@ export const reduceCrumbs = (
         index + 1
     );
 };
+
+export const updateSingleActiveLoopEvent = (state: ProcessInstanceEventState, action: PayloadAction<IProcessInstanceLoopEvent>) => {
+    if (!state.all.nestedEvents[action.payload.loopId]) return;
+
+    const iteration = state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber];
+    if (!iteration) {
+        state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber] = [action.payload];
+        return;
+    }
+
+    if (iteration.some((loopEvent) => loopEvent.id === action.payload.id)) {
+        state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber] = iteration
+            .map((loopEvent) => {
+                if (loopEvent.id === action.payload.id) return action.payload;
+                return loopEvent;
+            });
+        return;
+    }
+
+    iteration.push(action.payload);
+    iteration.sort(
+        (a: IProcessInstanceLoopEvent, b: IProcessInstanceLoopEvent) => new Date(a?.created).getTime() - new Date(b?.created).getTime(),
+    );
```

Comment:
function programming is preferred - e.g. 
const iterationEvents = [...state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber]]
const updatedEvents= [...iteration, action.payload]
const sortedNewEvents = updatedEvents.sort .............
state.all.nestedEvents[action.payload.loopId][action.payload.iterationNumber = [...sortedNewEvents]

Reply:
I don't see the point of spreading and copying an array if immer already does it under the hood, plus it's more concise this way